sonic_platform_base/sonic_xcvr/api/public/c_cmis.py

```diff
@@ -377,27 +438,11 @@ def get_transceiver_bulk_status(self):
         ========================================================================
         """
         trans_dom = super(CCmisApi,self).get_transceiver_bulk_status()
-        try:
-            trans_dom['bias_xi'] = self.vdm_dict['Modulator Bias X/I [%]'][1][0]
-            trans_dom['bias_xq'] = self.vdm_dict['Modulator Bias X/Q [%]'][1][0]
-            trans_dom['bias_xp'] = self.vdm_dict['Modulator Bias X_Phase [%]'][1][0]
-            trans_dom['bias_yi'] = self.vdm_dict['Modulator Bias Y/I [%]'][1][0]
-            trans_dom['bias_yq'] = self.vdm_dict['Modulator Bias Y/Q [%]'][1][0]
-            trans_dom['bias_yp'] = self.vdm_dict['Modulator Bias Y_Phase [%]'][1][0]
-            trans_dom['cd_shortlink'] = self.vdm_dict['CD high granularity, short link [ps/nm]'][1][0]
-            trans_dom['cd_longlink'] = self.vdm_dict['CD low granularity, long link [ps/nm]'][1][0]
-            trans_dom['dgd'] = self.vdm_dict['DGD [ps]'][1][0]
-            trans_dom['sopmd'] = self.vdm_dict['SOPMD [ps^2]'][1][0]
-            trans_dom['soproc'] = self.vdm_dict['SOP ROC [krad/s]'][1][0]
-            trans_dom['pdl'] = self.vdm_dict['PDL [dB]'][1][0]
-            trans_dom['osnr'] = self.vdm_dict['OSNR [dB]'][1][0]
-            trans_dom['esnr'] = self.vdm_dict['eSNR [dB]'][1][0]
-            trans_dom['cfo'] = self.vdm_dict['CFO [MHz]'][1][0]
-            trans_dom['tx_curr_power'] = self.vdm_dict['Tx Power [dBm]'][1][0]
-            trans_dom['rx_tot_power'] = self.vdm_dict['Rx Total Power [dBm]'][1][0]
-            trans_dom['rx_sig_power'] = self.vdm_dict['Rx Signal Power [dBm]'][1][0]
-        except KeyError:
-            pass
+
+        for vdm_key, trans_dom_key in VDM_KEY_TO_DB_KEY_PREFIX_MAP.items():
+            self._update_dict_if_vdm_key_exists(trans_dom, trans_dom_key, vdm_key, 0)
+
+        self._update_dict_if_vdm_key_exists(trans_dom, 'soproc', 'SOP ROC [krad/s]', 0)
```

Comment:
we should include soproc in the VDM_KEY_TO_DB_KEY_PREFIX_MAP dict.

Reply:
I have addressed this now.